python/lib/evaluations/dmod/evaluations/data_retriever/disk.py

```diff
@@ -93,7 +93,7 @@ def retrieve(self, *args, **kwargs) -> pandas.DataFrame:
                 )
                 frame = frame.join(constant_frame)
 
-            if self.definition.locations.from_field.lower() == "filename":
+            if self.definition.locations.from_field == "filename":
```

Comment:
It looks like `self.definition.locations` could be `None`. If that is the case, it seems like you would need something like:

 ```suggestion
            if self.definition.locations and self.definition.locations.from_field == "filename":
```

Reply:
Good catch - I put some extra checks in there to protect it and a similar issue.